src/mapping.ts

```diff
@@ -2,17 +2,11 @@ import { cosmos } from "@graphprotocol/graph-ts";
 import { MsgDelegate, MsgCoin, decodeMsgDelegate } from "./decoding";
 import { Delegation, Coin } from "../generated/schema";
 
-export function handleTx(transactionData: cosmos.TransactionData): void {
-  const id = `${transactionData.block.header.hash.toHexString()}-${transactionData.tx.index}`;
-  const messages = transactionData.tx.tx.body.messages;
+export function handleMsgDelegate(messageData: cosmos.MessageData): void {
+  const message = decodeMsgDelegate(messageData.message.value);
+  const id = `${messageData.block.header.hash.toHexString()}-${message.delegator_address}-${message.validator_address}`;
```

Comment:
Are we sure we won't have multiple delegations with the same validator and delegator address in one block?

Reply:
Not sure yet. We may need to add an index to messageData